### PR TITLE
feat(express_route_fix): change route to original path

### PR DESCRIPTION
**Pull request by haimrait**

### CODE REVIEW

File: src/runners/express.js [hunk 1]
```diff
@@ -38,6 +38,44 @@ function createRunner(req, startTime) {
...
+ * @return {Object} parametered path if not find
+ * return the last one.
+ */
+function findCalledParameteredPath(req) {
+    let matchedPath = req.route.path[req.route.path.length - 1];
+    req.route.path.forEach((parameteredPath) => {
+        const parameteredPathSplitted = parameteredPath.split('/');
+        const pathSplitted = req.path.split('/');
+        if (parameteredPathSplitted.length === pathSplitted.length) {
+            if (checkIfPathsAreEqual(parameteredPathSplitted, pathSplitted)) {
```
Comment (ranrib) [inline src/runners/express.js:71]: Can merge ifs

**Review by ranrib — DISMISSED** [review]

File: src/runners/express.js [hunk 2]
```diff
@@ -38,6 +38,44 @@ function createRunner(req, startTime) {
...
+ */
+function checkIfPathsAreEqual(parameteredPathSplitted, pathSplitted) {
+    for (let i = 0; i < pathSplitted.length; i += 1) {
+        if (parameteredPathSplitted[i] !== pathSplitted[i] &&
+            parameteredPathSplitted[i] &&
+            parameteredPathSplitted[i][0] !== ':') {
+            return false;
+        }
+    }
+
```
Comment (galbash) [inline src/runners/express.js:55]: there can be an index out of bounds exception here (make sure the length is the same first)

File: src/runners/express.js [hunk 3]
```diff
@@ -38,6 +38,44 @@ function createRunner(req, startTime) {
...
+ * @param {*} req - express http request object
+ * @return {Object} parametered path if not find
+ * return the last one.
+ */
+function findCalledParameteredPath(req) {
+    let matchedPath = req.route.path[req.route.path.length - 1];
+    req.route.path.forEach((parameteredPath) => {
+        const parameteredPathSplitted = parameteredPath.split('/');
+        const pathSplitted = req.path.split('/');
+        if (parameteredPathSplitted.length === pathSplitted.length) {
```
Comment (galbash) [inline src/runners/express.js]: should be inside the `checkIfPathsAreEqual` function

File: src/runners/express.js [hunk 4]
```diff
@@ -38,6 +38,44 @@ function createRunner(req, startTime) {
...
+}
+
+/**
+ * Find the parametered path that was called
+ * @param {*} req - express http request object
+ * @return {Object} parametered path if not find
+ * return the last one.
+ */
+function findCalledParameteredPath(req) {
+    let matchedPath = req.route.path[req.route.path.length - 1];
```
Comment (galbash) [inline src/runners/express.js:66]: isn't it better to return null and not set this tag if no path was found?

File: src/runners/express.js [hunk 5]
```diff
@@ -64,7 +102,12 @@ function finishRunner(expressEvent, res, req, startTime) {
     }
 
     if (req.route) {
-        eventInterface.addToMetadata(expressEvent, { route_path: req.route.path });
+        if (req.route.path instanceof Array) {
+            eventInterface.addToMetadata(expressEvent,
+                { route_path: findCalledParameteredPath(req) });
+        } else {
+            eventInterface.addToMetadata(expressEvent, { route_path: req.route.path });
```
Comment (galbash) [inline src/runners/express.js:107]: does this solve the case of nested routes?

**Review by galbash — CHANGES_REQUESTED** [review]

File: src/runners/express.js (hunk 2 above)
Comment (haimrait) [reply src/runners/express.js:55]: I check it in findCalledParameteredPath

File: src/runners/express.js (hunk 4 above)
Comment (haimrait) [reply src/runners/express.js:66]: But at Monday case you asked to set the last one if not found

File: src/runners/express.js (hunk 5 above)
Comment (haimrait) [reply src/runners/express.js:107]: Do we want to show the full path in this case?

**Review by galbash — APPROVED** [review]

**Pull request merged**